### PR TITLE
Add optimize clause in language_guide.md

### PR DESCRIPTION
**Pull request by sneaxiy**

### CODE REVIEW

File: doc/language_guide.md [hunk 1]
```diff
@@ -386,6 +386,69 @@ INTO evaluate_result_table;
 - *attr_expr* sets attributes when doing evaluation. You can set `validation.metrics` to indicate which metrics will be outputed to the result table, e.g. `validation.metrics="Accuracy,AUC"`, you can find more supported metric names for Keras models [here](https://www.tensorflow.org/api_docs/python/tf/keras/metrics) and XGBoost models [here](https://xgboost.readthedocs.io/en/latest/parameter.html).
 - *evaluate_result_table* is the result table that stores the evaluation results.
 
+## Optimization Syntax
+
+SQLFlow uses the `TO MAXIMIZE` and `TO MINIMIZE` clauses to describe and solve the math programming problems.
```
Comment (typhoonzero) [inline doc/language_guide.md]: math programming -> [Mathematical Programming](https://en.wikipedia.org/wiki/Mathematical_Programming)

File: doc/language_guide.md [hunk 2]
```diff
@@ -386,6 +386,69 @@ INTO evaluate_result_table;
...
+    SUM(x) <= c2 GROUP BY c1
+    x <= 3
+WITH
+    variables="x(c3)",
+    var_type="NonNegativeIntegers"
+USING glpk
+INTO my_db.my_result_table;
+```
+
+The objective expression following by `TO MAXIMIZE` means that we want to maximize the value of `SUM(x * c1)`.
```
Comment (typhoonzero) [inline doc/language_guide.md]: 1. use a list to describe the components in the statement
2. The Objective expression -> The `objective_expr`

File: doc/language_guide.md (hunk 1 above)
Comment (sneaxiy) [reply doc/language_guide.md]: Done.

File: doc/language_guide.md (hunk 2 above)
Comment (sneaxiy) [reply doc/language_guide.md]: Done.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**